pkg/config/run.go

```diff
@@ -68,13 +68,14 @@ type Step struct {
 }
 
 type StepCondition struct {
-	Config                    map[string][]interface{} `json:"config,omitempty"`
-	ConfigKey                 string                   `json:"configKey,omitempty"`
-	FilePattern               string                   `json:"filePattern,omitempty"`
-	FilePatternFromConfig     string                   `json:"filePatternFromConfig,omitempty"`
-	Inactive                  bool                     `json:"inactive,omitempty"`
-	NpmScript                 string                   `json:"npmScript,omitempty"`
-	CommonPipelineEnvironment map[string]interface{}   `json:"commonPipelineEnvironment,omitempty"`
+	Config                            map[string][]interface{} `json:"config,omitempty"`
+	ConfigKey                         string                   `json:"configKey,omitempty"`
+	FilePattern                       string                   `json:"filePattern,omitempty"`
+	FilePatternFromConfig             string                   `json:"filePatternFromConfig,omitempty"`
+	Inactive                          bool                     `json:"inactive,omitempty"`
+	NpmScript                         string                   `json:"npmScript,omitempty"`
+	CommonPipelineEnvironment         map[string]interface{}   `json:"commonPipelineEnvironment,omitempty"`
+	IfCommonPipelineEnvironmentFilled string                   `json:"ifCommonPipelineEnvironmentFilled,omitempty"`
```

Comment:
What about simplify the name a bit?

```suggestion
	pipelineEnvironmentFilled string                   `json:"pipelineEnvironmentFilled,omitempty"`
```

Reply:
I've made the suggested change. @anilkeshav27 we should then in the future rename the `commonPipelineEnvironment` condition as well.